configure.ac

```
@@ -202,7 +220,25 @@ set rb_dummy ${CC}
 rb_CC=$2
 AC_DEFUN([RUBY_CHECK_PROG_FOR_CC], [
     rb_prog=`echo "${rb_CC}" | sed "$2"`
-    AC_CHECK_PROG([$1], [$rb_prog], [$rb_prog])
+    rb_prog_found=no
+    # If the tool is not explicitly specified and the compiler
+    # is absolute, check the one next to the compiler.
+    AS_IF([test -z "$$1"], [
+        AS_CASE(["${rb_prog}"],
+        [/*], [
+            AC_MSG_CHECKING([for ${rb_prog}])
+            AS_IF([test -x "${rb_prog}"], [
+                rb_prog_found=yes
+                $1=${rb_prog}
+            ])
+            AC_MSG_RESULT([$rb_prog_found])
+        ])
+    ])
+    # Otherwise, try to find it in PATH with given program name
+    rb_prog=`echo "${rb_prog}" | sed 's/.*\///'`
+    AS_IF([test "${rb_prog_found}" = no], [
```

The basename of `rb_prog` is needed only if `rb_prog_found` is no.

```suggestion
    AS_IF([test "${rb_prog_found}" = no], [
        # Otherwise, try to find it in PATH with given program name
        rb_prog=`echo "${rb_prog}" | sed 's/.*\///'`
```